aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -343,28 +350,13 @@ module aptos_framework::fungible_asset {
                 )
             );
         });
-
-        // Cannot register hook for APT.
-        assert!(
-            object::address_from_constructor_ref(constructor_ref) != @aptos_fungible_asset,
-            error::permission_denied(EAPT_NOT_DISPATCHABLE)
-        );
-        assert!(
-            !object::can_generate_delete_ref(constructor_ref),
-            error::invalid_argument(EOBJECT_IS_DELETABLE)
-        );
+        register_dispatch_function_sanity_check(constructor_ref);
         assert!(
```

Comment:
why is this not a part of the sanity check function?

Reply:
Just for posterity's sake...
The assert below differs in each register function, based on the Struct being accessed.